File: 0078-subsets/0078-subsets.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """

        result = [[],nums]
        
        
        def combo(lister):
            for l in lister:
                copy = list(lister)
                copy.remove(l)
                copy.sort()
                if copy not in result:
                    result.append(copy)
                    combo(copy)
        combo(nums)
        return result

File: 0078-subsets/0078-subsets.java
class Solution {
    
    List<List<Integer>> answer = new ArrayList<>();
    
    public List<List<Integer>> subsets(int[] nums) {
        List<Integer> numbers = new ArrayList<>();
        
        for(int i: nums){
            numbers.add(i);
        }
        subSets(numbers, new ArrayList<Integer>());
        return answer;
    }
    
    public void subSets(List<Integer> nums, List<Integer> start){
        answer.add(start);
        if(nums.size() != 0){
        while(nums.size() > 0){
            ArrayList<Integer> copy = new ArrayList<>();
            copy.addAll(start);
            ArrayList<Integer> numbers = new ArrayList<>();
            numbers.addAll(nums);
            int value = nums.get(0);
            copy.add(value);
            numbers.remove(0);
            subSets(numbers, copy);
            nums.remove(0);
        }
        }
    }
}